src/object/srv_ec_aggregate.c

```diff
@@ -1327,8 +1327,10 @@ agg_peer_update_ult(void *arg)
 		D_FREE(ec_agg_in->ea_remove_recxs.ca_arrays);
 		D_FREE(ec_agg_in->ea_remove_eps.ca_arrays);
 	}
+
```

Comment:
I think after successful obj_req_create, we can probably assert that rpc != NULL but we should initialize it to NULL if we do that.

Reply:
I think the ABT_eventual_wait() call after the ult_create that spawns this function will block forever if the eventual isn't set prior to returning from this function.

Reply:
The correct fix, I believe, is to initialize the rpc pointer to NULL.  The goto is needed (the label can be changed to whatever you like).  The test for rpc != NULL after the goto is okay (since the pointer is properly initialized), and the ABT_eventual_set call is executed (which is necessary).

Reply:
Agree with this.  Just initialize rpc to NULL

Reply:
ok updated